x/fees/types/min_fee.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// MinFee contains the minimum amount of coins that should be paid as a fee for each message of the specified type sent
+type MinFee struct {
+	MessageType string    `json:"message_type" yaml:"message_type"`
+	Amount      sdk.Coins `json:"amount" yaml:"amount"`
+}
+
+func NewMinFee(messageType string, amount sdk.Coins) MinFee {
+	return MinFee{
+		MessageType: messageType,
+		Amount:      amount,
+	}
+}
+
+func (mf MinFee) String() string {
+	return fmt.Sprintf("Message Type: %s\nAmount: %s\n", mf.MessageType, mf.Amount)
+}
+
+// Validate check if the min fee parameters are valid
+func (mf MinFee) Validate() error {
+	if mf.MessageType == "" {
+		return fmt.Errorf("invalid minimum fee message type")
+	}
+
+	return nil
```

Comment:
We should also validate the `Amount` field by using the `IsValid` method of `sdk.Coins` here probably

Reply:
By default when creating a `Coin` or `Coins` type that check is performed automatically

Reply:
I think that check is performed only when creating the `Coin` or `Coins` using the `NewCoin` and `NewCoins` methods respectively. I don't think it's performed when the `Coins` are parsed from a JSON file. I think it might be better to add it for this reason anyway